src/main/java/pl/edu/agh/xp/advertisements/service/csv/CSVWriter.java

```diff
@@ -17,6 +16,10 @@ public void write(FileName filePath, Advertisement advertisement) {
         var fileToWrite = new File(filePath.getValue());
         if (!fileToWrite.exists()) {
             try {
+                var parent = fileToWrite.getParentFile();
+                if(!parent.exists()){
+                    parent.mkdirs();
```

Comment:
sprawdzałeś czy to działa rekurencyjnie? tzn jeśli parent nie istnieje to czy mkdirs() utworzy też parenta tego parenta itd?

Reply:
tak, działa, jak chcesz tylko jeden folder, to jest metoda mkdir()